worker/provisioner/provisioner_task.go

```diff
@@ -47,6 +47,7 @@ type ProvisionerTask interface {
 
 type MachineGetter interface {
 	Machine(names.MachineTag) (*apiprovisioner.Machine, error)
```

Comment:
There's no need to keep this single Machine() API call. There's only 2 call sites that aren't tests, here and jujud/agent. I'd consider perhaps defining Machines() API below using varidac args

Machines(...names.MachineTag) ([]apiprovisioner.MachineResult, error)

The remaining call site in jujud.agent can then just do a result[0] to get the single result. And the test usages in provisioner_test.go can use a helper if needed.

Reply:
updated.